std/numeric.d

```diff
@@ -171,8 +171,24 @@ struct CustomFloat(uint             precision,  // fraction bits (23 for float)
                    uint             exponentWidth,  // exponent bits (8 for float)  Exponent width
                    CustomFloatFlags flags,
                    uint             bias)
-if (((flags & flags.signed)  + precision + exponentWidth) % 8 == 0 &&
-    precision + exponentWidth > 0)
+    // Restrictions from bitfield
+if ((((flags & flags.signed) + precision + exponentWidth) == 8 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 16 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 32 ||
+    ((flags & flags.signed) + precision + exponentWidth) == 64 ||
+    precision == 64 &&  // due to CustomFloat!80 support hack, see below
+        ((flags & flags.signed) + exponentWidth == 8 ||
+        (flags & flags.signed) + exponentWidth == 16 ||
+        (flags & flags.signed) + exponentWidth == 32 ||
+        (flags & flags.signed) + exponentWidth == 64)) &&
+    // mantissa needs to fit into real mantissa
+    (precision <= real.mant_dig - 1 || precision == 64) &&
+    // exponent needs to fit into real exponent
+    1L << exponentWidth - 1 <= real.max_exp &&
+    // mantissa should have at least one bit
+    precision > 0 &&
```

Comment:
The current implementation assumes, that the mantissa has at least one bit. Therefore we make sure, that there is at least one.